src/main/groovy/com/blackducksoftware/integration/hub/detect/bomtool/yarn/YarnDependencyData.groovy

```diff
@@ -0,0 +1,92 @@
+/*
+ * hub-detect
+ *
+ * Copyright (C) 2018 Black Duck Software, Inc.
+ * http://www.blackducksoftware.com/
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package com.blackducksoftware.integration.hub.detect.bomtool.yarn
+
+class YarnDependencyData {
+
+    static Map<String, String> resolvedVersions
+
+    static Map<String, String> getYarnDataAsMap(List<String> inputLines) {
```

Comment:
Here the yarn lock is parsed into a map giving the resolved version for any components used in the project. This was done because yarn list returns fuzzy versions and the lock file is the most sensible place to find the resolved versions